docs/concepts/glossary.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Glossary
+nav-pos: 3
+nav-title: Glossary
+nav-parent_id: concepts
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+#### Flink Application Cluster
+
+A Flink Application Cluster is a dedicated [Flink Cluster](./glossary#flink-cluster) that only
+executes a single [Flink Job](./glossary#flink-job). The lifetime of the
+[Flink Cluster](./glossary#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*. Compare to
+[Flink Session Cluster](./glossary#flink-session-cluster).
+
+#### Flink Cluster
+
+The distributed system consisting of (typically) one Flink Master process and one or more Flink
```

Comment:
Suggestion: 
 * `The distributed system ...` -> `A distributed system ...`
 * `Flink Master process` -> `[Flink JobManager process](#flink-jobmanager-process)`
 * `Flink TaskManager processes` -> `[Flink TaskManager processes](#...)`

Reply:
2) and 3) done. 1) I am not sure.Maybe let's ask a native speaker. @sjwiesman what sounds more natural to you?

Reply:
"A distributed system"

Reply:
Also, drop the (typically) ?

Reply:
Standby JobManager Processes? I would considers these to be part of the Flink Cluster.

Reply:
Gotcha, maybe as a nit "The distributed system consisting of one Flink Master process, zero or more stand by job manager processes, and one or more Flink . . ."

Otherwise, lets get this merged in. We can always open follow up pull requests.